lib/tessel.js

```diff
@@ -441,6 +442,17 @@ function Tessel(options) {
     return new Pin(Object.assign({}, config, ToPortIdentity(index)));
   });
 
+  // REPL control:
+  // - If we detect that we're running on-board the Tessel, disable the repo
+  // - If the user specifies a repl option, use it no matter what
+  var tesselPathRegex = /^\/app\/remote-script/;
+  if (tesselPathRegex.test(__dirname)) {
+    this.repl = false;
+  }
+  if (typeof options.repl !== "undefined") {
+    this.repl = options.repl;
+  }
```

Comment:
It just occurred to me that maybe it's not obvious that `t2 run` and `t2 push` deploy to two different places in the file system. The former is deployed to ram (under /tmp/) and the latter to flash (under /app/), which is why it needs to default to true and why it needs to be changeable from tests